addons/spreadsheet/static/src/global_filters/helpers.js

```diff
@@ -229,6 +229,60 @@ const FILTERS_BEHAVIORS = {
         },
     ],
     boolean: [SET_OPERATORS_BEHAVIORS],
+    numeric: [
+        {
+            operators: ["=", "!=", ">", "<"],
+            defaultValue: { targetValue: undefined },
+            validateValue: (filterValue) => isNumericFilterValueValid(filterValue.targetValue),
+            validateDefaultValue: (filterValue) =>
+                isNumericFilterValueValid(filterValue.targetValue),
+            getSearchBarFacetValues: (env, filter, filterValue) => {
+                if (filterValue.targetValue === undefined) {
+                    return [];
+                }
+                return [`${filterValue.targetValue}`];
+            },
+            toDomain(fieldPath, filterValue) {
+                return new Domain([[fieldPath, filterValue.operator, filterValue.targetValue]]);
+            },
+            toCellValue(getters, filter, filterValue) {
+                return {
+                    value:
+                        filterValue.targetValue !== undefined ? `${filterValue.targetValue}` : "",
+                };
+            },
+        },
+        {
+            operators: ["between"],
+            defaultValue: { minimumValue: undefined, maximumValue: undefined },
+            validateValue: (filterValue) =>
+                isNumericFilterValueValid(filterValue.minimumValue) &&
+                isNumericFilterValueValid(filterValue.maximumValue),
+            validateDefaultValue: (filterValue) =>
+                isNumericFilterValueValid(filterValue.minimumValue) &&
+                isNumericFilterValueValid(filterValue.maximumValue),
+            getSearchBarFacetValues: (env, filter, filterValue) => {
+                if (
+                    filterValue.minimumValue === undefined &&
+                    filterValue.maximumValue === undefined
+                ) {
+                    return [];
+                }
+                return [`${filterValue.minimumValue} and ${filterValue.maximumValue}`];
```

Comment:
missing translation for 'and'. You can use `formatList`
```suggestion
                return [formatList([filterValue.minimumValue, filterValue.maximumValue])];
```